problems/problems.go

```diff
@@ -1,10 +1,46 @@
 package problems
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
 	"io/ioutil"
+	"log"
+	"net/http"
 	"os"
+	"strconv"
 )
 
+type CodeforcesProblem struct {
+	ContestID uint    `json:"contestId"`
+	Index     string  `json:"index"`
+	Name      string  `json:"name"`
+	Points    float32 `json:"points"`
+}
+
+type CodeforcesSubmission struct {
+	ID        uint              `json:"id"`
+	ContestID uint              `json:"contestId"`
+	Verdict   string            `json:"verdict"`
+	Problem   CodeforcesProblem `json:"problem"`
+}
+
+type CodeforcesSubmissionResponse struct {
+	Status string                 `json:"status"`
+	Result []CodeforcesSubmission `json:"result"`
+}
+
+// Problem describes a problem in our system.
+type Problem struct {
+	ID         uint
+	OwnerID    int
+	ExternalID string
+	Title      string
+	Source     string
+	HasSolved  bool
```

Comment:
Not sure if this has been defined on other places, but I was thinking that `HasBeenSolved` and `HasBeenTried` would sound better.

Reply:
About these fields, I'm thinking of removing them from Problem because there are two use cases:

1. When it's about a problem itself, not tied to any user, `HasBeenSolved` sounds better
2. When it's tied to a user, like in `/users/1/problems/3/`, `HasSolved` sounds better, IMHO

So, there should be a way to keep both version, or consolidate both. Suggestions are welcome.